WeakAuras/Profiling.lua

```diff
@@ -723,60 +525,273 @@ function RealTimeProfilingWindow:Init()
+  profilingFrame.mode = mode
+  UIDropDownMenu_SetText(profilingFrame.buttons.modeDropDown, modes[mode])
+  profilingFrame:ResetBars()
+  profilingFrame:RefreshBars(nil, true)
+end
+
+function WeakAurasProfilingMixin:InitModeDropDown()
+  local function Initializer(dropDown, level)
+		for i = 1, 2 do
+			local info = UIDropDownMenu_CreateInfo()
+			info.text = modes[i]
+			info.func = selectMode
+      info.checked = i == self.mode
+      info.arg1 = i
+			UIDropDownMenu_AddButton(info)
+		end
+	end
+
+	local dropDown = self.buttons.modeDropDown
+  UIDropDownMenu_SetWidth(dropDown, 120)
+  UIDropDownMenu_JustifyText(dropDown, "LEFT")
+  UIDropDownMenu_Initialize(dropDown, Initializer)
+end
+
+function WeakAurasProfilingMixin:OnShow()
+  --[[ 3x WeakAuras/Profiling.lua:593: Action[SetPoint] failed because[SetPoint would result in anchor family connection]: attempted from: WeakAurasProfilingFrame:SetPoint.
+  if WeakAurasSaved.RealTimeProfilingWindow then
+    self:SetPoint("TOPLEFT", UIParent, "TOPLEFT",
+                  WeakAurasSaved.RealTimeProfilingWindow.xOffset or 0,
+                  WeakAurasSaved.RealTimeProfilingWindow.yOffset or 0)
+  else
+    self:SetPoint("TOPLEFT", UIParent, "TOPLEFT")
   end
-  if currentProfileState == "encounter" then
-    self.encounterButton:SetText(L["Cancel"])
+  ]]
+end
+
+
+local lastRefresh
+function WeakAurasProfilingMixin:RefreshBars(_, force)
+  if force or (not lastRefresh or lastRefresh < GetTime() - 1) then
+    lastRefresh = GetTime()
   else
-    self.encounterButton:SetText(L["Next Encounter"])
+    return
   end
-  if currentProfileState == "profiling" then
-    self.toggleButton:SetText(L["Stop"])
-    self.combatButton:Hide()
-    self.encounterButton:Hide()
-    self.reportButton:Hide()
+
+  if not profileData.systems.time then
+    return
+  end
+
+  local data
+  if self.mode == 1 then -- auras
+    data = profileData.auras
+  elseif self.mode == 2 then -- systems
+    data = {}
+    for k, v in pairs(profileData.systems) do
+      if (k ~= "time" and k ~= "wa") then
+        local event = unitEventToMultiUnit(k)
+        if data[event] == nil then
+          data[event] = CopyTable(v)
+        else
+          if v.elapsed then
+            data[event].elapsed = (data[event].elapsed or 0) + v.elapsed
+          end
+          if v.spike then
+            data[event].spike = (data[event].spike or 0) + v.spike
+          end
+        end
+      end
+    end
+  end
+
+  local total = TotalProfileTime(data)
+  for _, name in ipairs(SortProfileMap(data)) do
+    if (name ~= "time" and name ~= "wa") then
+      local elapsed = data[name].elapsed
+      local pct = 100 * elapsed / total
+      local spike = data[name].spike
+      self:UpdateBar(name, elapsed, pct, spike)
+    end
+  end
+
+  self.bars:Sort()
+  if profileData.systems.wa then
+    local timespent = debugprofilestop() - profileData.systems.time.start
```

Comment:
If the profiling is stopped using the current time (via debugprofilestop) is wrong. Apart from the onUpdate bug above, this function is also called on switching the profiling mode.